test/lint/lint-submodule.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# This script checks for git modules
```

Comment:
post-hom: Probably should leave a sentence saying why it's erroring if it does and what the alternative is for implementers :)